### ISSUE TITLE
Support GitHub codespaces

### ISSUE THREAD

**Issue by tschm** [pola-rs/polars#9690]
### Problem description

Devcontainers support the creation of pull requests, see https://github.com/devcontainers

**Comment by ritchie46** [comment]
Could you elaborate your request? I don't understand.

**Comment by tschm** [comment]
I would recommend to start here: https://www.youtube.com/watch?v=-tQ2nxjqP6o

It's a fantastic concept!

The development is much simpler in particular for those that do not work on polars on a daily basis. You define the environment people use to code in. Nobody would be forced to use this approach though.

**Comment by eitsupi** [comment]
Indeed, the image used by default in Codespaces have Python and Rust installed (https://github.com/devcontainers/images/tree/main/src/universal), so I think it would be possible to develop this project without adding `devcontainer.json`.
(In other words, this repository already supports Codespaces)

**Comment by stinodego** [comment]
Yeah we should set this up. Will make it easier for people to contribute, especially since our dev environment is not that simple.

Will go on my backlog, unless you want to take a stab at this yourself.

**Comment by tschm** [comment]
I am happy to give it a go. It's an experiment for me. I have done it for Python projects before.

**Comment by tschm** [comment]
@stinodego I struggle to build polars-core. I am new to Polars and rust. Please have a look...

**Comment by eitsupi** [comment]
@tschm If this is out of your hands I think I can do it.
I have been using Dev Containers for the past few years and have set up a Dev Containers on https://github.com/PRQL/prql for example.

**Comment by tschm** [comment]
@eitsupi I would prefer giving this a spin tonight and submit a working pull request. You are most welcome to comment on https://github.com/tschm/polars/tree/main
